### DISCUSSION TITLE
Area64 - Middle Tennessee Intergroup

### DISCUSSION THREAD

**Discussion by lionssharedigital** [code4recovery/12-step-meeting-list#1080] [Q&A] [answered]
Area64 
https://www.area64assembly.org/ imports 4 data sources. The largest source is from AA Nashville with over 1k meetings. At the 
moment, when we try to run the refresh or import, we get a curl error.

The data source is:

https://aanashville.org/wp/wp-admin/admin-ajax.php?action=meetings

The error we get is:

WP_Error Object
(
    [errors] => Array
        (
            [http_request_failed] => Array
                (
                    [0] => cURL error 28: Connection timeout after 10000 ms
                )

        )

    [error_data] => Array
        (
        )

    [additional_data:protected] => Array
        (
        )

)

This was working in the past but now it's not. My guess is that it's just too much data and we need to increase the timeout setting. I don't have access to area64s server but could likely work with their person to fix this. 

I tried to delete the source and readd it but it failed with the same error.

**Comment by joshreisner** [comment]
Hm, not sure! It could be a network problem potentially between the two machines. 

I tried importing Nashville on our demo site and it works ok, leading me to think it's not a code problem but a site config problem.

The way the import process works it fetches all the data and puts it in a buffer, then imports in batches of 25 to avoid timeout. Sounds like it's timing out on that initial fetch.

You could debug by trying to run a PHP file that just fetches the data and see if it works?

At any rate, going to move this to a TSML discussion

**Reply by joshreisner** [reply]
looks like we have a [hard limit](https://github.com/code4recovery/12-step-meeting-list/blob/main/includes/admin_import.php#L165) of `30s` on imports, so that's another factor, does it take that long?

maybe a good PHP test would be something like:

```
// test.php
<?php
echo file_get_contents('https://aanashville.org/wp/wp-admin/admin-ajax.php?action=meetings');
```

**Reply by lionssharedigital** [reply]
Good idea. I tried this and get a 504 Gateway timeout. It sounds like the two servers are unable to connect to each other.

**Reply by joshreisner** [reply]
ok good. best guess is it's some well-meaning person at one of the hosting companies blocking it because it looked suspicious. i think if you contact them you can persuade them to reopen the connection

**Reply by marvabban** [reply]
Hello,
I have the same issue. The feed I am importing is valid but the plugin times-out when importing, same error as above. I created a file with the file_get_contents function, and it hangs for 60s and fails. I can't get an error from chrome developer tools. Please go ahead and test the file I created, https://aasask.org/stoon-load-test.php
My host says the address is not blocked by them, they suggested an .htaccess file might be to blame? I'm wondering if anyone knows how I could fix this?
Thanks,
Mark

**Reply by joshreisner** [reply]
hey @marvabban thanks - i actually can't load that URL - it takes about 60s for me in my browser and ends with an empty response (white screen) - you might use https://meetingguide.org/validate or https://codebeautify.org/jsonvalidator to test whether third party sites can see the address

**Reply by marvabban** [reply]
Hi @joshreisner , thanks for your reply. 
The feed I am trying to import is https://aasaskatoon.org/wp-admin/admin-ajax.php?action=meetings and it is valid and should load in a browser.
The site I am trying to import to is https://aasask.org but I get a time out. I created a test page, https://aasask.org/stoon-load-test.php, with only code suggested in an earlier comment `<?php
echo file_get_contents('https://aanashville.org/wp/wp-admin/admin-ajax.php?action=meetings');` and it times-out as well, so I think there is something on my site, aasask.org, that is not allowing connection.
Thanks,
Mark

**Reply by joshreisner** [reply, marked as answer]
thanks @marvabban for the clarification yes the feed works great. i'm pretty sure then the problem is with your host, they said "the address is not blocked by them" - i'm guessing that all outbound requests are blocked, not just that specific one. [here's another relatively small open feed](https://westalaa.org/wp-admin/admin-ajax.php?action=meetings) that you could try importing to confirm 

i would tell them that outbound requests appear to be blocked. perhaps there's a non-json way to confirm? maybe try this? https://webmasters.stackexchange.com/questions/92355/how-can-i-check-if-my-hosting-provider-blocked-outbound-traffic-to-some-ips

because you confirmed that it doesn't work with pure php i think you can be sure that the problem is not wordpress (or `.htaccess`) related
